server/src/main/java/io/crate/replication/logical/action/TransportDropSubscriptionAction.java

```diff
@@ -103,28 +101,23 @@ protected ClusterBlockException checkBlock(DropSubscriptionRequest request, Clus
     }
 
     /**
-     * Removes the REPLICATION_SUBSCRIPTION_NAME index setting from all relevant to subscription indices.
+     * Removes the REPLICATION_SUBSCRIPTION_NAME index setting from all indices included by the subscription.
      */
-    private Metadata.Builder removeSubscriptionSetting(ImmutableOpenMap<String, IndexMetadata> indices,
-                                                       Metadata.Builder mdBuilder,
-                                                       @Nonnull String subscriptionToDrop) {
-        Iterator<IndexMetadata> indicesIterator = indices.valuesIt();
-        while (indicesIterator.hasNext()) {
-            IndexMetadata indexMetadata = indicesIterator.next();
-            var settings = indexMetadata.getSettings();
-            var subscriptionName = REPLICATION_SUBSCRIPTION_NAME.get(settings); // Can be null.
-            if (subscriptionToDrop.equals(subscriptionName)) {
-                var settingsBuilder = Settings.builder().put(settings);
-                settingsBuilder.remove(REPLICATION_SUBSCRIPTION_NAME.getKey());
-                mdBuilder.put(
-                    IndexMetadata
-                        .builder(indexMetadata)
-                        .settingsVersion(1 + indexMetadata.getSettingsVersion())
-                        .settings(settingsBuilder)
-                );
-            }
+    private void removeSubscriptionSetting(Subscription subscription,
+                                           Metadata metadata,
+                                           Metadata.Builder mdBuilder) {
+        for (var relationName : subscription.relations().keySet()) {
+            IndexMetadata indexMetadata = metadata.index(relationName.indexNameOrAlias());
+            assert indexMetadata != null : "Cannot resolve indexMetadata for relation=" + relationName;
+            var settingsBuilder = Settings.builder().put(indexMetadata.getSettings());
+            settingsBuilder.remove(REPLICATION_SUBSCRIPTION_NAME.getKey());
+            mdBuilder.put(
+                IndexMetadata
+                    .builder(indexMetadata)
+                    .settingsVersion(1 + indexMetadata.getSettingsVersion())
```

Comment:
I wonder if we should compute version once outside of the loop  - looks like we can have "holes" in version as final value will be `oldValue + iterationsCount` - not a problem probably but just nicer to have it sequential?